### PR TITLE
Implement Sorting and Pagination for IRS and MDA Point Plans

### LINKED ISSUE #968: RVL-994 - Implement Sorting and Pagination for IRS and MDA Point Plans under Monitor Section
The IRS and MDA Point plans under the Monitor section do not have the sorting on the headers and the pagination features implemented. The user would like to have these features similar to how they are implemented on other pages of the WebUI.

*Acceptance Criteria*
1. Login into the reveal WebUI
2. Go to Monitor > IRS and to Monitor > MDA Point sections
3. The displayed plans should have ability to sort on the column headers and should have pagination implemented.

### PR DESCRIPTION
**Pull request by ciremusyoka**
Closes #968 
Adds pagination and sorting on table headers for `Monitor > IRS` and `Monitor > MDA Point` pages by making use of [drill down table](https://www.npmjs.com/package/@onaio/drill-down-table) package.

### CODE REVIEW

**Comment by ciremusyoka** [comment]
screenShot
Screenshot from 2020-07-07 18-46-56
Screenshot from 2020-07-07 18-46-43

File: src/containers/pages/GenericPlansList/index.tsx [hunk 1]
```diff
@@ -69,31 +74,73 @@ export const GenericPlansList = (props: GenericPlanListProps & RouteComponentPro
...
+    DrillDownTableProps<Dictionary>,
+    | 'columns'
+    | 'data'
+    | 'loading'
+    | 'loadingComponent'
+    | 'renderInBottomFilterBar'
+    | 'renderInTopFilterBar'
+    | 'useDrillDown'
+    | 'renderNullDataComponent'
+  > = {
```
Comment (moshthepitt) [inline src/containers/pages/GenericPlansList/index.tsx:119]: I have seen this used a couple of places.  Could we create this as a type alias for convenient re-use?  Not a fan of all these repeated magic strings

File: src/containers/pages/IRS/plans/tests/__snapshots__/index.test.tsx.snap [hunk 2]
```diff
@@ -159,6 +159,246 @@ exports[`components/IRS Reports/IRSPlansList renders plan definition list correc
 </h3>
 `;
 
-exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: table headers 1`] = `null`;
+exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: table headers 1`] = `
+<div
```
Comment (moshthepitt) [inline src/containers/pages/IRS/plans/tests/__snapshots__/index.test.tsx.snap:163]: Questions:

- why is it changing from null to something else? (looks like previously the null was intentional)
- Does this snapshot need to be this large just to show that headers were rendered?

File: src/containers/pages/IRS/plans/tests/__snapshots__/index.test.tsx.snap [hunk 3]
```diff
@@ -159,6 +159,246 @@ exports[`components/IRS Reports/IRSPlansList renders plan definition list correc
...
+`;
+
+exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: the full rendered rows tr index 0 1`] = `"TitleDate CreatedStart DateEnd DateStatus"`;
+
+exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: the full rendered rows tr index 1 1`] = `"IRS 2019-09-05 TEST2019-09-052019-09-052019-09-25retired"`;
+
+exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: the full rendered rows tr index 2 1`] = `"IRS 2019-08-30m2019-08-302019-08-302019-09-14active"`;
 
-exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: table rows 1`] = `null`;
+exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: the full rendered rows tr index 3 1`] = `"Berg Namibia 20192019-08-292019-08-292019-09-18active"`;
```
Comment (moshthepitt) [inline src/containers/pages/IRS/plans/tests/__snapshots__/index.test.tsx.snap:404]: Is the change from null to something else expected?

File: src/containers/pages/IRS/plans/tests/index.test.tsx [hunk 4]
```diff
@@ -70,15 +71,34 @@ describe('components/IRS Reports/IRSPlansList', () => {
...
-    await new Promise(resolve => setImmediate(resolve));
-    wrapper.update();
+    await act(async () => {
+      await new Promise(resolve => setImmediate(resolve));
+      wrapper.update();
+    });
     expect(toJson(wrapper.find('BreadcrumbItem li'))).toMatchSnapshot('breadcrumbs');
     expect(toJson(wrapper.find('h3.page-title'))).toMatchSnapshot('page title');
-    expect(toJson(wrapper.find('.thead .tr .th'))).toMatchSnapshot('table headers');
-    expect(toJson(wrapper.find('.tbody .tr .td'))).toMatchSnapshot('table rows');
```
Comment (moshthepitt) [inline src/containers/pages/IRS/plans/tests/index.test.tsx:78]: Why?

File: src/containers/pages/MDAPoint/plans/tests/__snapshots__/index.test.tsx.snap [hunk 5]
```diff
@@ -139,164 +139,237 @@ exports[`components/MDA Reports/MDAPlansList renders plan definition list correc
...
-    key="2"
-  >
-    2019-08-29
-  </td>,
-  <td
-    className="listview-th"
-    key="3"
-  >
-    2019-09-18
-  </td>,
```
Comment (moshthepitt) [inline src/containers/pages/MDAPoint/plans/tests/__snapshots__/index.test.tsx.snap:294]: I am nervous here.  We are removing the data in the row from the snapshot.  Why?

**Comment by moshthepitt** [comment]
I think the snapshots here are larger than they could be.  Please consult: https://kentcdodds.com/blog/effective-snapshot-testing/

File: src/containers/pages/MDAPoint/plans/tests/__snapshots__/index.test.tsx.snap (hunk 5 above)
Comment (ciremusyoka) [reply src/containers/pages/MDAPoint/plans/tests/__snapshots__/index.test.tsx.snap:294]: The snapshot is too big i decided to just check the number of  items being rendered.

File: src/containers/pages/IRS/plans/tests/index.test.tsx (hunk 4 above)
Comment (ciremusyoka) [reply src/containers/pages/IRS/plans/tests/index.test.tsx:78]: > expect(toJson(wrapper.find('.thead .tr .th'))).toMatchSnapshot('table headers');

I believe ^^ is still present.

File: src/containers/pages/IRS/plans/tests/__snapshots__/index.test.tsx.snap (hunk 2 above)
Comment (ciremusyoka) [reply src/containers/pages/IRS/plans/tests/__snapshots__/index.test.tsx.snap]: I think it should not have been `null` if you look at the [test](https://github.com/onaio/reveal-frontend/blob/87dc5b5186a5f9e059377fe134d0adf9a5c9a0f2/src/containers/pages/IRS/plans/tests/index.test.tsx#L77), at that point table headers should be rendered.

File: src/containers/pages/IRS/plans/tests/__snapshots__/index.test.tsx.snap (hunk 3 above)
Comment (ciremusyoka) [reply src/containers/pages/IRS/plans/tests/__snapshots__/index.test.tsx.snap]: These are different snapshots

File: src/containers/pages/IRS/plans/tests/index.test.tsx (hunk 4 above)
Comment (ciremusyoka) [reply src/containers/pages/IRS/plans/tests/index.test.tsx:78]: for this `expect(toJson(wrapper.find('.tbody .tr .td'))).toMatchSnapshot('table rows');` the current table we are using is giving large snapshots for this so i opted to remove it and check if the right number of rows were rendered

**Comment by Naima-Bashir** [comment]
How to test: search bar functionality, sorting functionality
image

**Pull request merged**